File: gaussdb/src/test/java/com/dbx/agent/gaussdb/GaussdbAgentTest.java
package com.dbx.agent.gaussdb;

import com.dbx.agent.BaseDatabaseAgent;
import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class GaussdbAgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new GaussdbAgent();
    }

    @Override
    protected String resultSetSql() {
        return "CALL sample_proc()";
    }

    @Test
    void agentExtendsBaseDatabaseAgent() {
        Assertions.assertTrue(BaseDatabaseAgent.class.isAssignableFrom(GaussdbAgent.class));
    }
}

File: gaussdb/src/test/kotlin/com/dbx/agent/gaussdb/GaussdbAgentTest.kt
package com.dbx.agent.gaussdb

import com.dbx.agent.test.JdbcAgentFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class GaussdbAgentTest {
    @Test
    fun `executes non select statements that return result sets`() {
        val agent = GaussdbAgent()
        setPrivateConnection(agent, JdbcAgentFake.connection())

        val result = agent.executeQuery("CALL sample_proc()", null)

        assertEquals(listOf("VALUE"), result.columns)
        assertEquals(listOf(listOf("row-value")), result.rows)
        assertEquals(listOf("execute"), JdbcAgentFake.calls)
    }
}
